### PR TITLE
Feat add agent manager

### PR DESCRIPTION
**Pull request by iFurySt**
modified includes:

- add agent manager to manage all agents;
- extract the host ssh port to prevent conflict.
- clean all containers with the prefix sandbox-

### CODE REVIEW

**Review by rbren — APPROVED** [review]

**Comment by iFurySt** [comment]
image
image

do we decide to merge this? it can't create a new container if previous container is still exist, the hardcode host port 2222 will cause the problem.

**Comment by rbren** [comment]
Yes, I think this fixes a lot of problems with the SSHBox.

**Comment by iFurySt** [comment]
i've resolved the conflicts, appreciate it if you have time to merge it.

File: frontend/src/state/settingsSlice.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export const settingsSlice = createSlice({
     setByKey: (state, action) => {
       const { key, value } = action.payload;
       state[key] = value;
+      console.log("setByKey", key, value);
```
Comment (rbren) [inline frontend/src/state/settingsSlice.ts:21]:
```suggestion
```

File: opendevin/sandbox/ssh_box.py [hunk 2]
```diff
@@ -33,21 +34,35 @@
 
 
 class DockerSSHBox(Sandbox):
-    closed = False
+    instance_id: str
+    container_image: str
+    container_name_prefix = 'sandbox-'
```
Comment (rbren) [inline opendevin/sandbox/ssh_box.py:39]:
```suggestion
    container_name_prefix = 'opendevin-sandbox-'
```

File: opendevin/sandbox/exec_box.py [hunk 3]
```diff
@@ -32,21 +32,32 @@
 
 
 class DockerExecBox(Sandbox):
-    closed = False
+    instance_id: str
+    container_image: str
+    container_name_prefix = 'sandbox-'
```
Comment (rbren) [inline opendevin/sandbox/exec_box.py:37]:
```suggestion
    container_name_prefix = 'opendevin-sandbox-'
```

**Review by rbren — APPROVED** [review]

**Comment by rbren** [comment]
A few quick notes:
* I changed the prefix to `opendevin-sandbox-`, which should be a little safer
* If we instantiate multiple sandboxes in the future, the cleanups might fight each other...may need to rethink that eventually

**Comment by rbren** [comment]
Will merge once tests pass

**Comment by katmai** [comment]
i'm getting an error when starting. noted it here: https://github.com/OpenDevin/OpenDevin/issues/1050

**Pull request merged**